src/structures/cl_args.cpp

```diff
@@ -26,13 +27,27 @@ void update_host(Servers& servers, const std::string& value) {
     host = value.substr(index + 1);
   }
 
+  // remove any trailing slash
+  if (host.find('/') == host.length() - 1) {
+    host.pop_back();
+  }
+
+  // pull out a path if any and append a trailing slash for query building
+  index = host.find('/');
+  if (index != std::string::npos) {
+    path = host.substr(index + 1) + "/";
```

Comment:
Passing an empty host used to be OK during update and generate a "clean" error upon connection error. On the other hand, this will break:
```
$ vroom -a "" ...
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
```

Wrapping lines 30-40 in a `if (!host.empty())` clause should do the trick?

Reply:
jep, fixed